File: app/src/main/java/com/jcs/where/features/gourmet/restaurant/list/filter/AreaFilterFragment.java
package com.jcs.where.features.gourmet.restaurant.list.filter;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

import com.jcs.where.R;
import com.jcs.where.api.response.area.AreaResponse;
import com.jcs.where.base.BaseFragment;
import com.jcs.where.view.empty.EmptyView;

import java.util.ArrayList;

/**
 * Created by Wangsw  2021/3/29 15:55.
 * 餐厅商业区筛选
 */
public class AreaFilterFragment extends BaseFragment {


    private final ArrayList<AreaResponse> dataList = new ArrayList<>();
    private AreaFilterAdapter mAdapter;
    private EmptyView mEmptyView;

    public AreaFilterFragment getInstance(ArrayList<AreaResponse> dataList) {
        this.dataList.addAll(dataList);
        return new AreaFilterFragment();
    }

    private RecyclerView contentRv;

    @Override
    protected int getLayoutId() {
        return R.layout.single_recycler_view;
    }

    @Override
    protected void initView(View view) {
        contentRv = view.findViewById(R.id.content_rv);
        mEmptyView = new EmptyView(requireContext());
    }

    @Override
    protected void initData() {
        mAdapter = new AreaFilterAdapter();
        mAdapter.setEmptyView(mEmptyView);
        mAdapter.setNewInstance(dataList);
        contentRv.setAdapter(mAdapter);
    }

    @Override
    protected void bindListener() {

    }


}

File: app/src/main/java/com/jiechengsheng/city/features/gourmet/restaurant/list/filter/area/AreaFilterFragment.kt
package com.jiechengsheng.city.features.gourmet.restaurant.list.filter.area

import android.view.View

import com.jiechengsheng.city.base.mvp.BaseMvpFragment
import com.jiechengsheng.city.view.empty.EmptyView
import androidx.recyclerview.widget.RecyclerView
import com.jiechengsheng.city.R
import com.jiechengsheng.city.api.response.area.AreaResponse
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.listener.OnItemClickListener
import org.greenrobot.eventbus.EventBus
import com.jiechengsheng.city.base.BaseEvent

/**
 * Created by Wangsw  2021/3/29 15:55.
 * 餐厅商业区筛选
 */
class AreaFilterFragment : BaseMvpFragment<AreaFilterPresenter?>(), AreaFilterView, OnItemClickListener {


    private lateinit var mAdapter: AreaFilterAdapter
    private lateinit var mEmptyView: EmptyView
    private lateinit var contentRv: RecyclerView

    override fun getLayoutId() = R.layout.single_recycler_view

    override fun initView(view: View) {
        contentRv = view.findViewById(R.id.content_rv)
        contentRv.setVerticalScrollBarEnabled(true)
        mEmptyView = EmptyView(requireContext())
    }

    override fun initData() {
        presenter = AreaFilterPresenter(this)
        mAdapter = AreaFilterAdapter()
        mAdapter!!.setEmptyView(mEmptyView!!)
        contentRv!!.adapter = mAdapter
        mAdapter!!.setOnItemClickListener(this)
    }

    override fun bindListener() {}
    override fun loadOnVisible() {
        presenter!!.getAreasList()
    }

    override fun bindList(response: List<AreaResponse>) {
        mAdapter!!.setNewInstance(response.toMutableList())
    }

    override fun onItemClick(adapter: BaseQuickAdapter<*, *>, view: View, position: Int) {
        val list: List<AreaResponse> = mAdapter!!.data
        for (i in list.indices) {
            val data = list[i]
            data.nativeIsSelected = position == i
        }
        mAdapter!!.notifyDataSetChanged()
        val areaData = mAdapter!!.data[position]
        EventBus.getDefault().post(BaseEvent(areaData))
    }
}